src/graphics.c

```diff
@@ -49,48 +61,50 @@ struct graphics * graphics_create( struct graphics *parent )
 	return g;
 }
 
-int graphics_write( struct graphics_command *command ) {
-    int window = -1;
-    char* str;
-    struct graphics_color c;
-
-    while (command && command->type) {
-        switch (command->type) {
-            case GRAPHICS_WINDOW:
-                window = command->args[0];
-                if (window < 0 || window >= current->window_count) {
-                    return -1;
-                }
-                break;
-            case GRAPHICS_COLOR:
-                c.r = command->args[0];
-                c.g = command->args[1];
-                c.b = command->args[2];
-                c.a = 0;
-                graphics_fgcolor(current->windows[window], c);
-                break;
-            case GRAPHICS_RECT:
-                graphics_rect(current->windows[window], command->args[0], command->args[1], command->args[2], command->args[3]);
-                break;
-            case GRAPHICS_CLEAR:
-                graphics_clear(current->windows[window], command->args[0], command->args[1], command->args[2], command->args[3]);
-                break;
-            case GRAPHICS_LINE:
-                graphics_line(current->windows[window], command->args[0], command->args[1], command->args[2], command->args[3]);
-                break;
-            case GRAPHICS_TEXT:
-                str = (char*)command->args[2];
-                int i;
-                for (i = 0; str[i]; i++) {
-                    graphics_char(current->windows[window],	command->args[0]+i*8, command->args[1], str[i]);
-                }
-                break;
-            default:
-                break;
+int graphics_device_write( struct device *d, void *buffer, int size ) {
+    printf("I AM TRYING OK %x\n", d);
```

Comment:
Temporary - for some reason d will always be 0 here, even if this function is invoked through `d->write(d, buffer, size);`